src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -27,6 +27,12 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; 'handler' functions for HTTP endpoints
 
+(defn handle-server-error
```

Comment:
Oooh, good find.

I think the implementation of that is better but since that is also tied to a [jruby specific wrapper](https://github.com/puppetlabs/puppet-server/blob/0f8cd3602517477a02140bfd961de6a785800a04/src/clj/puppetlabs/puppetserver/jruby_request.clj#L43-L51) and would need to refactor out the generally useful part of the wrapper, and that this was intended as a small maint patch, I would lean towards not consolidating them at this time.

I think that same code is also _very_ applicable to [this handler](https://github.com/puppetlabs/puppet-server/pull/998/files#diff-bb1d8f2d842888913c49c1dddc70a5eaR64) that I just wrote as well. Since that's a bit more meaty ticket I _think_ I'd prefer for refactoring as part of that or creating a ticket for the refactoring.

One of the tickets in my working-remote-queue was [TK-214](https://tickets.puppetlabs.com/browse/TK-214) which would consolidate ring middleware/handler helpers. I _think_ my preference would be to either merge this PR and #998 and then work on TK-214 next, or merge this ticket and then do a little bit of refactoring to use the code you found in #998 if we won't get to TK-214 soon.


And of course, when I say preference above I mean more of a musing on your last sentence, which I took as an earnest question -- just let me know if you meant it more as an indirect imperative.  (;